File: 0027-remove-element/0027-remove-element.py
class Solution:
    def removeElement(self, nums: List[int], val: int) -> int:
        l, r = 0, len(nums)-1
        while l <= r:
            while l < len(nums) and nums[l] != val:
                l += 1
            while r >= 0 and nums[r] == val:
                r -= 1
            if l < r:
                nums[l], nums[r] = nums[r], nums[l]
        return l

File: 0027-remove-element/0027-remove-element.java
class Solution {
  public int removeElement(int[] nums, int val) {
    int l = 0;
    int r = nums.length - 1;
    int temp;
    while (l <= r) {
      while (l < nums.length && nums[l] != val) l++;
      while (r >= 0 && nums[r] == val) r--;
      if (l < r) {
        temp = nums[l];
        nums[l] = nums[r];
        nums[r] = temp;
      }
    }
    return l;
  }
}